app/Models/Ldap.php

```diff
@@ -119,6 +120,25 @@ public static function findAndBindUserLdap($username, $password)
 
         \Log::debug('Filter query: '.$filterQuery);
 
+	// userDn should be independent from baseDn (maybe you want to search in >=2 subtrees)
+	// -> better ask LDAP for user dn, that's why it is for
+	if ($settings->is_ad != '1') {
+		$userresults = ldap_search($connection, $baseDn, $filterQuery);
+		$userentries = ldap_get_entries($connection, $userresults);
+		// Can be empty if user does not exist
+		if ( $userentries["count"] > 0 ) {
+			$dn = $userentries[0]['dn'];
+			if ( $dn ) {
+				\Log::debug('User dn is: ' .$dn);
+				$userDn = $dn;
+			} else {
+				\Log::debug('User dn is empty.');
+			}
+		} else {
+			\Log::debug('Status of LDAP entries for user ' .$username. ': no result.');
```

Comment:
If you do the above-listed change, you'll probably want to output the 'failed' `$userentries["count"]` here so they can troubleshoot that there were either no results, or more than one result, I figure?

Reply:
Although that case should always be 0 in reality I added a further debugging parameter, s. also https://github.com/snipe/snipe-it/pull/11715
You can merge the other MR and close this one, or the other way round, I wouldn't care too much.

BTW: Since the last changes in develop LDAP login does not seem to work any more ("SAML page requested, but SAML does not seem to enabled."). We have no SAML active (maybe this should be opened as separated issue?).

Reply:
You can safely ignore the SAML logging. It's simply because it's in a service provider so it gets loaded on every request. It has nothing at all to do with LDAP syncing or auth.

Reply:
"Since the last changes in develop LDAP login does not seem to work any more"
Sorry, was my fault. Since bookworm I have to activate "Allow invalid SSL Certificate" although we have an official letsencrypt certificate with our LDAP servers. Something seems having become stricter.

Reply:
Happy to help you along with your specific issues - come ping us on Discord, maybe we can figure something out. Sometimes it's having a modern CA bundle.

Reply:
Has nothing to do with snipe-it, but just FYI: When ldap.so is linked towards libgnutls, LDAP is much stricter than when linked against libssl. On systems where a patched php-ldap is installed, there is no certificate problem.